### PR TITLE
Pagination refactor

### PR DESCRIPTION
**Pull request by coakenfold**
The PR for `FW-1481-kids-phrases` is a bit too large so I'm pulling out features that can be PRed separately.

- Pagination using Data, Presentation & Container technique. Paged items are set via props.children.
- Adding jsDocs to Frontend. Run via `npm run docs` and it will output to `frontend/qa/Documentation`. `qa/Documentation` is gitignored

### CODE REVIEW

File: frontend/app/assets/javascripts/components/Pagination/PaginationPresentation.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+}) {
+  return (
+    <div className="Pagination">
+      <div className="row">
+        <div className="col-xs-12">{children}</div>
+      </div>
+
+      <div className="row PrintHide Pagination__container">
+        <div className="col-md-7 col-xs-12 Pagination__pager">
+          <Pagination
```
Comment (dy2288) [inline frontend/app/assets/javascripts/components/Pagination/PaginationPresentation.js:47]: Is there a reason not to use react-paginate directly here and cut the middle-man of: https://github.com/First-Peoples-Cultural-Council/fv-web-ui/blob/master/frontend/app/assets/javascripts/views/components/Navigation/Pagination/index.js

File: frontend/app/assets/javascripts/components/Pagination/PaginationData.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+const { func, number } = PropTypes
+PaginationData.propTypes = {
+  children: func.isRequired,
+  page: number,
+  pageSize: number,
+  pageCount: number,
+  resultsCount: number,
+}
+PaginationData.defaultProps = {
+  page: 1,
```
Comment (dy2288) [inline frontend/app/assets/javascripts/components/Pagination/PaginationData.js:57]: one thing to consider (maybe?) is that in the backend when querying with the "currentPageIndex" property pages start at 0. This was previously adjusted in the "fetcher", not sure if it is relevant here or not: https://github.com/First-Peoples-Cultural-Council/fv-web-ui/blob/f6be172848a01b1e3d221e63e24c3301223b28ed/frontend/app/assets/javascripts/views/pages/explore/dialect/media/index.js#L188 

But something you should confirm.

File: frontend/app/assets/javascripts/components/Pagination/PaginationPresentation.js (hunk 1 above)
Comment (coakenfold) [reply frontend/app/assets/javascripts/components/Pagination/PaginationPresentation.js:47]: An omission due to time constraints. 

I was converting https://github.com/First-Peoples-Cultural-Council/fv-web-ui/blob/master/frontend/app/assets/javascripts/views/hoc/grid-list/with-pagination.js and didn't examine the internals too closely. 

I'll make the change, should be easy to pull out.

File: frontend/app/assets/javascripts/components/Pagination/PaginationData.js (hunk 2 above)
Comment (coakenfold) [reply frontend/app/assets/javascripts/components/Pagination/PaginationData.js:57]: I'll have to look into this. I was able to paginate as expected during ad hoc testing but need to confirm how the code making the request is doing it.

Comment (coakenfold) [reply frontend/app/assets/javascripts/components/Pagination/PaginationData.js:57]: Found 3 locations in master where `page` is being reduced to accommodate a zero index:
- https://github.com/First-Peoples-Cultural-Council/fv-web-ui/blob/master/frontend/app/assets/javascripts/views/pages/explore/dialect/learn/songs-stories/view.js#L129
- https://github.com/First-Peoples-Cultural-Council/fv-web-ui/blob/master/frontend/app/assets/javascripts/views/pages/explore/dialect/media/index.js#L188
- https://github.com/First-Peoples-Cultural-Council/fv-web-ui/blob/master/frontend/app/assets/javascripts/views/components/Editor/SelectMediaComponent.js#L296

I think adjusting `page` should be the responsibility of the component using the new `Pagination`. For example, there are other examples where components are using `page` as is to update the url, eg:
https://github.com/First-Peoples-Cultural-Council/fv-web-ui/blob/master/frontend/app/assets/javascripts/views/pages/explore/dialect/Recorders/index.js#L240

**Comment by coakenfold** [comment]
> Is there a component that is or will be using this refactor?

Yes over in `FW-1481-kids-phrases`. There are 2 new components that are handling the Kids Words & Phrase views:
- https://github.com/First-Peoples-Cultural-Council/fv-web-ui/blob/FW-1481-kids-phrases/frontend/app/assets/javascripts/components/KidsWordsByCategory/KidsWordsByCategoryContainer.js#L38
- https://github.com/First-Peoples-Cultural-Council/fv-web-ui/blob/FW-1481-kids-phrases/frontend/app/assets/javascripts/components/KidsPhrasesByCategory/KidsPhrasesByCategoryContainer.js#L39

**Pull request merged**